src/main/kotlin/org/rust/ide/utils/ExprUtils.kt

```diff
@@ -0,0 +1,212 @@
+        else -> null
+    }
+}
+
+/**
+ * Enum class representing unary operator in rust.
+ */
+enum class UnaryOperator {
+    REF, // `&a`
+    REF_MUT, // `&mut a`
+    DEREF, // `*a`
+    MINUS, // `-a`
+    NOT, // `!a`
+    BOX, // `box a`
+}
+
+/**
+ * Operator of current psi node with unary operation.
+ *
+ * The result can be [REF] (`&`), [REF_MUT] (`&mut`),
+ * [DEREF] (`*`), [MINUS] (`-`), [NOT] (`!`),
+ * [BOX] (`box`) or `null` if none of these.
+ */
+val RsUnaryExpr.operatorType: UnaryOperator?
+    get() = when {
+        this.and != null -> UnaryOperator.REF
+        this.mut != null -> UnaryOperator.REF_MUT
+        this.mul != null -> UnaryOperator.DEREF
+        this.minus != null -> UnaryOperator.MINUS
+        this.excl != null -> UnaryOperator.NOT
+        this.box != null -> UnaryOperator.BOX
+        else -> null
+    }
+
+/**
+ * Simplifies a boolean expression if can.
+ *
+ * @return `(expr, result)` where `expr` is a resulting expression,
+ *         `result` is true if an expression was actually simplified.
+ */
+fun RsExpr.simplifyBooleanExpression(): Pair<RsExpr, Boolean> {
+    if (this is RsLitExpr)
+        return this to false
+    return this.evalBooleanExpression()?.let {
+        createPsiElement(project, it) to true
+    } ?: when (this) {
+        is RsBinaryExpr -> {
+            val (leftExpr, leftSimplified) = left.simplifyBooleanExpression()
+            val (rightExpr, rightSimplified) = right!!.simplifyBooleanExpression()
+            if (leftExpr is RsLitExpr) {
+                leftExpr.boolLiteral?.let {
+                    when (this.operatorType) {
+                        ANDAND ->
+                            when (it.text) {
+                                "true" -> return rightExpr to true
+                                "false" -> return createPsiElement(project, "false") to true
+                            }
+                        OROR ->
+                            when (it.text) {
+                                "true" -> return createPsiElement(project, "true") to true
+                                "false" -> return rightExpr to true
+                            }
+                    }
+                    {}
+                }
+            }
+            if (rightExpr is RsLitExpr) {
+                rightExpr.boolLiteral?.let {
+                    when (this.operatorType) {
+                        ANDAND ->
+                            when (it.text) {
+                                "true" -> return leftExpr to true
+                                "false" -> return createPsiElement(project, "false") to true
+                            }
+                        OROR ->
+                            when (it.text) {
+                                "true" -> return createPsiElement(project, "true") to true
+                                "false" -> return leftExpr to true
+                            }
+                    }
+                    {}
```

Comment:
I think it would be slightly better to write `Unit`, which is a synonym of Rust's `()`. `{}` is a trivial function.